deluge/tests/test_web_api.py

```diff
@@ -9,35 +9,34 @@
 import json
 from io import BytesIO
 
+import pytest
+import pytest_twisted
 from twisted.internet import defer, reactor
-from twisted.python.failure import Failure
 from twisted.web.client import Agent, FileBodyProducer
 from twisted.web.http_headers import Headers
 from twisted.web.static import File
 
 import deluge.component as component
-from deluge.ui.client import client
 
 from . import common
 from .common_web import WebServerTestBase
 
 common.disable_new_release_check()
 
 
-class WebAPITestCase(WebServerTestBase):
-    def test_connect_invalid_host(self):
-        d = self.deluge_web.web_api.connect('id')
-        d.addCallback(self.fail)
-        d.addErrback(self.assertIsInstance, Failure)
-        return d
+class TestWebAPI(WebServerTestBase):
+    @pytest.mark.xfail(reason='This just logs an error at the moment.')
```

Comment:
The test claims it should raise an exception, but it just logs the error right now. Don't know if we want to update the behavior.